src/flash.js

```diff
@@ -621,6 +634,162 @@ class Flash extends FakeEventTarget implements IEngine {
   get availableBuffer(): number {
     return NaN;
   }
+
+  /**
+   * Sets an image to be shown while the video is downloading, or until the user hits the play button.
+   * @param {string} poster - The image url to be shown.
+   * @returns {void}
+   * @public
+   */
+  set poster(poster: string): void {}
+
+  /**
+   * Gets an image to be shown while the video is downloading, or until the user hits the play button.
+   * @returns {poster} - The image url.
+   * @public
+   */
+  get poster(): string {
+    return '';
+  }
+
+  /**
+   * Specifies if and how the author thinks that the video should be loaded when the page loads.
+   * @param {string} preload - The preload value.
+   * @public
+   * @returns {void}
+   */
+  set preload(preload: string): void {}
+
+  /**
+   * Gets the preload value of the video element.
+   * @returns {string} - The preload value.
+   * @public
+   */
+  get preload(): string {
+    return 'none';
+  }
+
+  /**
+   * Set if the video will automatically start playing as soon as it can do so without stopping.
+   * @param {boolean} autoplay - The autoplay value.
+   * @public
+   * @returns {void}
+   */
+  set autoplay(autoplay: boolean): void {}
+
+  /**
+   * Gets the autoplay value of the video element.
+   * @returns {boolean} - The autoplay value.
+   * @public
+   */
+  get autoplay(): boolean {
+    return false;
+  }
+
+  /**
+   * Set to specifies that video controls should be displayed.
+   * @param {boolean} controls - the controls value.
+   * @public
+   * @returns {void}
+   */
+  set controls(controls: boolean): void {}
+
+  /**
+   * Gets the controls value of the video element.
+   * @returns {boolean} - The controls value.
+   * @public
+   */
+  get controls(): boolean {
+    return false;
+  }
+
+  /**
+   * Set to specifies that the video will start over again, every time it is finished.
+   * @param {boolean} loop - the loop value.
+   * @public
+   * @returns {void}
+   */
+  set loop(loop: boolean) {}
+
+  /**
+   * Gets the loop value of the video element.
+   * @returns {boolean} - The loop value.
+   * @public
+   */
+  get loop(): boolean {
+    return false;
+  }
+
+  isPictureInPictureSupported(): void {}
```

Comment:
Shouldn’t this return Boolean?

Reply:
You're right I saw it now it came from youtube engine and it defined like that.